interbtc-spec/docs/source/spec/nomination.rst

```diff
@@ -5,84 +5,61 @@ Vault Nomination
-#. Vault replacement is disallowed for Operators with nominated collateral. Otherwise, Security Assumptions 1 and 2 would be violated.
-#. The nominated DOT:
-
-   #. Is locked on the parachain
-
-   #. Cannot be withdrawn by the operating Vault
-
-   #. Is capped at a fraction of the Vault’s deposited collateral (Max Nomination Ratio). This prevents the Operator from withdrawing its entire collateral and only exposing Nominators to economic risk, or stealing without liquidation consequences. This means that an Operator can only withdraw collateral as long as the fraction of nominated collateral does not exceed the threshold cap. Capping Nominator collateral also prevents Operators being “outnumbered” by Nominators and their relative fee earnings being marginalized.
-
-#. Liquidation slashing is handled as follows.
-
-   #. In case the collateral managed by the Operator falls below the liquidation threshold, the Operator and Nominators are slashed proportionally to their collateral.
-   
-   #. In case the Operator steals Bitcoin deposited at its address, its collateral is used to cover as much of the slashed amount as possible. If the Operator’s collateral was not enough to cover the entire amount, the Nominators are slashed proportionally for the remaining amount.
-
-#. Collateral withdrawals are first requested and then executed. A withdrawal request:
-
-   #. Decreases the issuable interbtc capacity.
-   
-   #. May be cancelled if not executed. The amount in the cancelled withdrawal request becomes backing collateral again.
-   
-   #. Is subject to a window of delay (unbonding period) that allows Nominators and the Operator to react.
+#. Vaults must choose to opt in to the Nomination protocol.
+#. Nominators select a Vault to which they can delegate collateral. They will earn a fraction of any rewards generated by the Vault.
+#. Vault replacement is disallowed with nominated collateral. Otherwise, Security Assumptions 1 and 2 would be violated.
+#. The nominated collateral:
 
-      #. Operator Unbonding Period. This window is longer, because Nominators are assumed to be mostly offline.
+   #. Cannot be withdrawn by the Vault,
+   #. Is locked on the parachain,
+   #. Is capped at a fraction of the Vault’s deposited collateral (:ref:`maxNominationRatio`). 
       
-      #. Nominator Unbonding Period. This window is shorter, because the Vault Operator is assumed to always be online.
+#. Liquidation slashing is handled as follows:
 
-#. Collateral withdrawals are subject to the following restrictions.
+   #. In case the collateral managed by the Vault falls below the liquidation threshold, the Vault and Nominators are slashed proportionally to their collateral.
+   #. In case the Vault steals Bitcoin deposited at its address, the Vault and Nominators are slashed proportionally to their collateral.
```

Comment:
This is currently how it is implemented, but remember that it is proportional and we ensure that the Vault has a bigger stake in the pool. If we wish to change it to your suggestion we would need to model what this looks like in the staking pool.

Reply:
How can we reflect that the vault is slashed first? This is part of the current protocol definition so it should work like that. Essentially, the specification should not be updated because of an implementation change, but rather, we need to think about how the implementation can be updated so it is in line with the specification again.

Reply:
I see that it is easier to implement both the same way, but at this point, the vault is clearly at fault. The main problem I see is in the following case:

Incentive to steal BTC:
- Assume no nomination: At Vault collateralization below 100%, a Vault should steal Bitcoin.
- Assume slash Vault first, Nominators after with 25% of collateral nominated: At Vault collateralization below 100%, a Vault should steal Bitcoin. Effective threshold when Vault has an incentive to steal Bitcoin: 100% + (100% * 25%) = 125% collateralization. In case of theft, the vault is slashed all collateral, the nominators are slashed all collateral since we slash up to the secure collateral threshold. However, if the Vault steals Bitcoin at collateralization of 187.5% (i.e., 150% + (150% * 25%)), all of the vault's collateral are slashed and none of the nominators collateral is slashed. Normally, the vault should not be motivated to steal but it might be the case if e.g., you modify my example from the comment above (exchange rate drops, not yet reflected on chain, nominators are offline and cannot react, new exchange rate would bring combined collaterealization below 125% (such that vault's future collateral is below 100%)). In this case, the vault should steal BTC but in this case, we would only slash the vault for this.
- Assume proportional slashing with 25% of collateral nominated: At Vault collateralization below 100%, a Vault should steal Bitcoin. Effective threshold when Vault has an incentive to steal Bitcoin: 100% + (100% * 25%) = 125% collateralization. In case of theft, the vault is slashed all collateral, the nominators are slashed all collateral since we slash up to the secure collateral threshold. If the Vault steals Bitcoin at collateralization of 187.5% (i.e., 150% + (150% * 25%)), the vault's and nominators' collateral are slashed proportionally such that 150%/187.5% = 80% of the collateral is slashed from both the vault and its nominators. Normally, the vault should not be motivated to steal but it might be the case if e.g., you modify my example from the comment above (exchange rate drops, not yet reflected on chain, nominators are offline and cannot react, new exchange rate would bring combined collaterealization below 125% (such that vault's future collateral is below 100%))

Comparing the proportional slashing with the Vault's first slashing (assuming fully nominated Vaults):
1. The effective collateralization rate at which Vault's should steal Bitcoin increases from 100% to 125%.
2. At the 125% collateralization rate, the punishment for Vaults and Nominators is the same: both get 100% of their collateral slashed.
3. Up to collateralization rate of 187.5% (secure collateral  / (secure collateral + (secure collateral * max nomination ratio))), Vaults get 100% of their collateral slashed and Nominators none under the Vault first slashing approach. Below 187.5%, Nominators share of slashed collateral increases. In the proportional slashing, there is no such differentiation.

So changing to proportional slashing does not change the effective threshold to steal, but it changes the way that stealing is handled such that Vaults might have a higher motivation to steal at higher collateralization thresholds. This is a practical concern when e.g., exchange rate drops are relayed with 1-hour delay so that vault clients can react in advance.

![image](https://user-images.githubusercontent.com/14966470/125195808-ac8ebd80-e257-11eb-8195-5702ae2da249.png)